<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Plot multi-GWAS/QTL or multi-ancestry fine-mapping results — plot_locus_multi • echoplot</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Plot multi-GWAS/QTL or multi-ancestry fine-mapping results — plot_locus_multi"><meta property="og:description" content="Plot multi-GWAS/QTL or multi-ancestry(i.e. trans-ethnic) fine-mapping
 results generated by tools like PAINTOR."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echoplot</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/echoplot.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/RajLabMSSM/echoplot/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot multi-GWAS/QTL or multi-ancestry fine-mapping results</h1>
    <small class="dont-index">Source: <a href="https://github.com/RajLabMSSM/echoplot/blob/HEAD/R/plot_locus_multi.R" class="external-link"><code>R/plot_locus_multi.R</code></a></small>
    <div class="hidden name"><code>plot_locus_multi.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Plot multi-GWAS/QTL or multi-ancestry(i.e. trans-ethnic) fine-mapping
 results generated by tools like PAINTOR.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">plot_locus_multi</span><span class="op">(</span></span>
<span>  <span class="va">dat_ls</span>,</span>
<span>  <span class="va">LD_ls</span>,</span>
<span>  <span class="va">locus_dir</span>,</span>
<span>  conditions <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">dat_ls</span><span class="op">)</span>,</span>
<span>  show_plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>dat_ls</dt>
<dd><p>A named list of <a href="https://Rdatatable.gitlab.io/data.table/reference/data.table.html" class="external-link">data.table</a>s generated by
<code>echolocatoR::finemap_loci</code>.</p></dd>


<dt>LD_ls</dt>
<dd><p>A named list of link disequilibrium (LD) matrices 
(one per item in <code>dat_ls</code>).</p></dd>


<dt>locus_dir</dt>
<dd><p>Storage directory to use.</p></dd>


<dt>conditions</dt>
<dd><p>Conditions to group <code>dat_ls</code> by.
The length of <code>conditions</code> must equal the number of
 items in <code>dat_ls</code>.</p></dd>


<dt>show_plot</dt>
<dd><p>Print plot to screen.</p></dd>


<dt>verbose</dt>
<dd><p>Print messages.</p></dd>


<dt>...</dt>
<dd><p>Arguments passed on to <code><a href="plot_locus.html">plot_locus</a></code></p><dl><dt><code>dataset_type</code></dt>
<dd><p>Dataset type (e.g. "GWAS" or "eQTL").</p></dd>

    <dt><code>color_r2</code></dt>
<dd><p>Whether to color data points (SNPs)
by how strongly they correlate with the lead SNP 
(i.g. LD measured in terms of r2).</p></dd>

    <dt><code>finemap_methods</code></dt>
<dd><p>Fine-mapping methods to plot tracks for, 
where the y-axis show the Posterior Probabilities (PP) 
of each SNP being causal.</p></dd>

    <dt><code>track_order</code></dt>
<dd><p>The order in which tracks should appear
(from top to bottom).</p></dd>

    <dt><code>track_heights</code></dt>
<dd><p>The height of each track (from top to bottom).</p></dd>

    <dt><code>plot_full_window</code></dt>
<dd><p>Include a track with a Manhattan plot of the full
GWAS/eQTL locus (not just the zoomed-in portion).</p></dd>

    <dt><code>dot_summary</code></dt>
<dd><p>Include a dot-summary plot that highlights the
Lead, Credible Set, and Consensus SNPs.</p></dd>

    <dt><code>qtl_suffixes</code></dt>
<dd><p>If columns with QTL data is included in <code>dat</code>,
you can indicate which columns those are with one or more string suffixes
(e.g. <code>qtl_suffixes=c(".eQTL1",".eQTL2")</code> to use the columns 
"P.QTL1", "Effect.QTL1", "P.QTL2", "Effect.QTL2").</p></dd>

    <dt><code>mean.PP</code></dt>
<dd><p>Include a track showing mean Posterior Probabilities (PP) 
averaged across all fine-mapping methods.</p></dd>

    <dt><code>sig_cutoff</code></dt>
<dd><p>Filters out SNPs to plot based on an (uncorrected)
p-value significance cutoff.</p></dd>

    <dt><code>gene_track</code></dt>
<dd><p>Include a track showing gene bodies.</p></dd>

    <dt><code>max_transcripts</code></dt>
<dd><p>Maximum number of transcripts per gene.</p></dd>

    <dt><code>tx_biotypes</code></dt>
<dd><p>Transcript biotypes to include in the gene model track. 
By default (<code>NULL</code>), all transcript biotypes will be included.
See <a href="get_tx_biotypes.html">get_tx_biotypes</a> for a full list of 
all available biotypes</p></dd>

    <dt><code>point_size</code></dt>
<dd><p>Size of each data point.</p></dd>

    <dt><code>point_alpha</code></dt>
<dd><p>Opacity of each data point.</p></dd>

    <dt><code>snp_group_lines</code></dt>
<dd><p>Include vertical lines to help highlight 
SNPs belonging to one or more of the following groups:
 Lead, Credible Set, Consensus.</p></dd>

    <dt><code>labels_subset</code></dt>
<dd><p>Include colored shapes and RSID labels
to help highlight SNPs belonging to one or more of the following groups:
 Lead, Credible Set, Consensus.</p></dd>

    <dt><code>xtext</code></dt>
<dd><p>Include x-axis title and text for each track
(not just the lower-most one).</p></dd>

    <dt><code>show_legend_genes</code></dt>
<dd><p>Show the legend for the <code>gene_track</code>.</p></dd>

    <dt><code>zoom_exceptions_str</code></dt>
<dd><p>Names of tracks to exclude when zooming.</p></dd>

    <dt><code>xgr_libnames</code></dt>
<dd><p>Passed to <a href="XGR_plot.html">XGR_plot</a>.
Which XGR annotations to check overlap with.
For full list of libraries see
<a href="http://XGR_r-forge.r-project.org/#annotations-at-the-genomic-region-level" class="external-link">
 here</a>.
 Passed to the <code>RData.customised</code> argument in <a href="https://rdrr.io/pkg/XGR/man/xRDataLoader.html" class="external-link">xRDataLoader</a>.
 Examples:</p><ul><li><p>"ENCODE_TFBS_ClusteredV3_CellTypes"</p></li>
<li><p>"ENCODE_DNaseI_ClusteredV3_CellTypes"</p></li>
<li><p>"Broad_Histone"</p></li>
</ul></dd>

    <dt><code>xgr_n_top</code></dt>
<dd><p>Passed to <a href="XGR_plot.html">XGR_plot</a>.
Number of top annotations to be plotted
(passed to <a href="https://rdrr.io/pkg/echoannot/man/XGR_filter_sources.html" class="external-link">XGR_filter_sources</a> and then 
<a href="https://rdrr.io/pkg/echoannot/man/XGR_filter_assays.html" class="external-link">XGR_filter_assays</a>).</p></dd>

    <dt><code>nott_epigenome</code></dt>
<dd><p>Include tracks showing brain cell-type-specific 
epigenomic data from 
<a href="https://doi.org/10.1126/science.aay0793" class="external-link">Nott et al. (2019)</a>.</p></dd>

    <dt><code>nott_regulatory_rects</code></dt>
<dd><p>Include track generated by 
<a href="https://rdrr.io/pkg/echoannot/man/NOTT2019_epigenomic_histograms.html" class="external-link">NOTT2019_epigenomic_histograms</a>.</p></dd>

    <dt><code>nott_show_placseq</code></dt>
<dd><p>Include track generated by
<a href="https://rdrr.io/pkg/echoannot/man/NOTT2019_plac_seq_plot.html" class="external-link">NOTT2019_plac_seq_plot</a>.</p></dd>

    <dt><code>nott_binwidth</code></dt>
<dd><p>When including Nott et al. (2019)
epigenomic data in the track plots,
adjust the bin width of the histograms.</p></dd>

    <dt><code>nott_bigwig_dir</code></dt>
<dd><p>Instead of pulling Nott et al. (2019)
 epigenomic data
from the <em>UCSC Genome Browser</em>, use a set of local bigwig files.</p></dd>

    <dt><code>roadmap</code></dt>
<dd><p>Find and plot annotations from Roadmap.</p></dd>

    <dt><code>roadmap_n_top</code></dt>
<dd><p>Passed to <a href="ROADMAP_plot.html">ROADMAP_plot</a>.
Number of top annotations to be plotted
(passed to <a href="https://rdrr.io/pkg/echoannot/man/ROADMAP_query.html" class="external-link">ROADMAP_query</a>).</p></dd>

    <dt><code>roadmap_query</code></dt>
<dd><p>Only plot annotations from Roadmap whose
metadata contains a string or any items from  a list of strings
(e.g. <code>"brain"</code> or <code>c("brain","liver","monocytes")</code>).</p></dd>

    <dt><code>credset_thresh</code></dt>
<dd><p>The minimum fine-mapped posterior probability
 for a SNP to be considered part of a Credible Set.
For example, <code>credset_thresh=.95</code> means that all Credible Set SNPs
will be 95% Credible Set SNPs.</p></dd>

    <dt><code>facet_formula</code></dt>
<dd><p>Formula to facet plots by.
See <a href="https://ggplot2.tidyverse.org/reference/facet_grid.html" class="external-link">facet_grid</a> for details.</p></dd>

    <dt><code>save_plot</code></dt>
<dd><p>Save plot as RDS file.</p></dd>

    <dt><code>plot_format</code></dt>
<dd><p>Format to save plot as 
when saving with <a href="https://ggplot2.tidyverse.org/reference/ggsave.html" class="external-link">ggsave</a>.</p></dd>

    <dt><code>save_RDS</code></dt>
<dd><p>Save the tracks as an RDS file 
(<em>Warning</em>: These plots take up a lot disk space).</p></dd>

    <dt><code>return_list</code></dt>
<dd><p>Return a named list with each track as a separate plot
(default: <code>FALSE</code>). If <code>TRUE</code>, will return a merged plot using
<a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a>.</p></dd>

    <dt><code>dat</code></dt>
<dd><p>Data to query transcripts with.</p></dd>

    <dt><code>LD_reference</code></dt>
<dd><p>LD reference to use:</p><ul><li><p>"1KGphase1" : 1000 Genomes Project Phase 1 (genome build: hg19).</p></li>
<li><p>"1KGphase3" : 1000 Genomes Project Phase 3 (genome build: hg19).</p></li>
<li><p>"UKB" : Pre-computed LD from a British
European-decent subset of UK Biobank.
<em>Genome build</em> : hg19</p></li>
<li><p>"&lt;vcf_path&gt;" : User-supplied path to a custom VCF file 
to compute LD matrix from.<br><em>Accepted formats</em>: <em>.vcf</em> / <em>.vcf.gz</em> / <em>.vcf.bgz</em><br><em>Genome build</em> : defined by user with <code>target_genome</code>.</p></li>
<li><p>"&lt;matrix_path&gt;" : User-supplied path to a pre-computed LD matrix   
<em>Accepted formats</em>:  <em>.rds</em> / <em>.rda</em> / <em>.csv</em> / 
<em>.tsv</em> / <em>.txt</em><br><em>Genome build</em> : defined by user with <code>target_genome</code>.</p></li>
</ul></dd>

    <dt><code>conda_env</code></dt>
<dd><p>Conda environments to search in. 
If <code>NULL</code> (default), will search all conda environments.</p></dd>

    <dt><code>nThread</code></dt>
<dd><p>Number of threads to parallelize over.</p></dd>

    <dt><code>LD_matrix</code></dt>
<dd><p>LD matrix.</p></dd>

    <dt><code>zoom</code></dt>
<dd><p>Zoom into the center of the locus when plotting
(without editing the fine-mapping results file).
You can provide either:</p><ul><li><p>The size of your plot window in terms of basepairs
(e.g. <code>zoom=50000</code> for a 50kb window).</p></li>
<li><p>How much you want to zoom in (e.g. <code>zoom="1x"</code>
for the full locus, <code>zoom="2x"</code>
for 2x zoom into the center of the locus, etc.).</p></li>
</ul><p>You can pass a list of window sizes (e.g. <code>c(50000,100000,500000)</code>)
to automatically generate
multiple views of each locus.
This can even be a mix of different style inputs: e.g.
 <code>c("1x","4.5x",25000)</code>.</p></dd>

    <dt><code>genomic_units</code></dt>
<dd><p>Which genomic units to return window limits in.</p></dd>

    <dt><code>density_adjust</code></dt>
<dd><p>Passed to <code>adjust</code> argument in 
<a href="https://ggplot2.tidyverse.org/reference/geom_density.html" class="external-link">geom_density</a>.</p></dd>

    <dt><code>strip.text.y.angle</code></dt>
<dd><p>Angle of the y-axis facet labels.</p></dd>

    <dt><code>dpi</code></dt>
<dd><p>dpi to use for raster graphics</p></dd>

    <dt><code>height</code></dt>
<dd><p>height (defaults to the height of current
  plotting window)</p></dd>

    <dt><code>width</code></dt>
<dd><p>width (defaults to the width of current
  plotting window)</p></dd>

    <dt><code>consensus_thresh</code></dt>
<dd><p>The minimum number of fine-mapping tools 
in which a SNP is in the Credible Set in order to be included in the 
"Consensus_SNP" column.</p></dd>

  
</dl></dd>

</dl></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">locus_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,<span class="fu">echodata</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/echodata/man/locus_dir.html" class="external-link">locus_dir</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co">#### Make dat_ls ####</span></span></span>
<span class="r-in"><span><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu">echodata</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/echodata/man/filter_snps.html" class="external-link">filter_snps</a></span><span class="op">(</span><span class="fu">echodata</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/echodata/man/BST1.html" class="external-link">BST1</a></span>, bp_distance <span class="op">=</span> <span class="fl">10000</span><span class="op">)</span> </span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> FILTER:: Filtering by SNP features.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + FILTER:: Post-filtered data: 114 x 26</span>
<span class="r-in"><span><span class="va">dat_ls</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>gwas1<span class="op">=</span><span class="va">dat</span>, gwas2<span class="op">=</span><span class="va">dat</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co">#### Make LD_ls ####</span></span></span>
<span class="r-in"><span><span class="va">LD_matrix</span> <span class="op">&lt;-</span> <span class="fu">echodata</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/echodata/man/BST1_LD_matrix.html" class="external-link">BST1_LD_matrix</a></span></span></span>
<span class="r-in"><span><span class="va">LD_ls</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>ancestry1<span class="op">=</span><span class="va">LD_matrix</span>, ancestry2<span class="op">=</span><span class="va">LD_matrix</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co">#### Make plot ####</span></span></span>
<span class="r-in"><span><span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu">plot_locus_multi</span><span class="op">(</span>dat_ls <span class="op">=</span> <span class="va">dat_ls</span>, </span></span>
<span class="r-in"><span>                              LD_ls <span class="op">=</span> <span class="va">LD_ls</span>,</span></span>
<span class="r-in"><span>                              locus_dir <span class="op">=</span> <span class="va">locus_dir</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> LD_matrix detected. Coloring SNPs by LD with lead SNP.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filling r/r2 NAs with 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> LD_matrix detected. Coloring SNPs by LD with lead SNP.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filling r/r2 NAs with 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> +-------- Locus Plot:  BST1 --------+</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + support_thresh = 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Calculating mean Posterior Probability (mean.PP)...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + 4 fine-mapping methods used.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + 8 Credible Set SNPs identified.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + 4 Consensus SNPs identified.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Filling NAs in CS cols with 0.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Filling NAs in PP cols with 0.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ++ echoplot:: GWAS full window track</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ++ echoplot:: GWAS track</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ++ echoplot:: Merged fine-mapping track</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Melting PP and CS from 5 fine-mapping methods.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + echoplot:: Constructing SNP labels.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding SNP group labels to locus plot.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ++ echoplot:: Adding Gene model track.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting dat to GRanges object.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> max_transcripts= 1 . </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1  transcripts from  1  genes returned.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Fetching data...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> OK</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Parsing exons...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> OK</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Defining introns...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> OK</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Defining UTRs...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> OK</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Defining CDS...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> OK</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> aggregating...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Constructing graphics...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Adding vertical lines to highlight SNP groups.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> +&gt;+&gt;+&gt;+&gt;+ zoom =  1x  +&lt;+&lt;+&lt;+&lt;+</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + echoplot:: Get window suffix...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + echoplot:: Removing GWAS full window track @ zoom=1x</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Removing subplot margins...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Reordering tracks...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Ensuring last track shows genomic units.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Aligning xlimits for each subplot...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Checking track heights...</span>
<span class="r-plt img"><img src="plot_locus_multi-1.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder, Jack Humphrey, Towfique Raj.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

