<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Plot annotation peaks: XGR — XGR_plot • echoplot</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Plot annotation peaks: XGR — XGR_plot"><meta property="og:description" content="Import annotation from XGR, filter to only those within the range of
dat, and then plot the peaks.
If the annotation has already been downloaded previously, it will be reused."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">echoplot</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/echoplot.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/RajLabMSSM/echoplot/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot annotation peaks: XGR</h1>
    <small class="dont-index">Source: <a href="https://github.com/RajLabMSSM/echoplot/blob/HEAD/R/XGR_plot.R" class="external-link"><code>R/XGR_plot.R</code></a></small>
    <div class="hidden name"><code>XGR_plot.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Import annotation from <em>XGR</em>, filter to only those within the range of
<code>dat</code>, and then plot the peaks.
If the annotation has already been downloaded previously, it will be reused.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">XGR_plot</span><span class="op">(</span></span>
<span>  <span class="va">dat</span>,</span>
<span>  lib_name <span class="op">=</span> <span class="st">"ENCODE_TFBS_ClusteredV3_CellTypes"</span>,</span>
<span>  locus_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  palette <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Spectral"</span>, <span class="st">"BrBG"</span>, <span class="st">"PiYG"</span>, <span class="st">"PuOr"</span><span class="op">)</span>,</span>
<span>  fill_var <span class="op">=</span> <span class="st">"Assay"</span>,</span>
<span>  facet_var <span class="op">=</span> <span class="st">"Source"</span>,</span>
<span>  geom <span class="op">=</span> <span class="st">"density"</span>,</span>
<span>  n_top <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  adjust <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>  force_new <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  show_plot <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  nThread <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>dat</dt>
<dd><p><a href="https://Rdatatable.gitlab.io/data.table/reference/data.table.html" class="external-link">data.table</a> with at least the following columns:</p><dl><dt>SNP</dt>
<dd><p>SNP RSID</p></dd>

<dt>CHR</dt>
<dd><p>chromosome</p></dd>

<dt>POS</dt>
<dd><p>position</p></dd>


</dl></dd>


<dt>lib_name</dt>
<dd><p>Which XGR annotations to check overlap with.
For full list of libraries see
<a href="http://XGR_r-forge.r-project.org/#annotations-at-the-genomic-region-level" class="external-link">
 here.</a>
 Passed to the <code>RData.customised</code> argument in <a href="https://rdrr.io/pkg/XGR/man/xRDataLoader.html" class="external-link">xRDataLoader</a>.</p></dd>


<dt>locus_dir</dt>
<dd><p>Locus-specific directory.</p></dd>


<dt>palette</dt>
<dd><p>Palette name to be passed to <code>name</code> argument in 
<a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html" class="external-link">brewer.pal</a>.</p></dd>


<dt>fill_var</dt>
<dd><p>Fill variable.</p></dd>


<dt>facet_var</dt>
<dd><p>Row facet variable.</p></dd>


<dt>geom</dt>
<dd><p>Plot type ("density", or "histogram").</p></dd>


<dt>n_top</dt>
<dd><p>Number of top annotations to be plotted
(passed to <a href="https://rdrr.io/pkg/echoannot/man/XGR_filter_sources.html" class="external-link">XGR_filter_sources</a> and then
<a href="https://rdrr.io/pkg/echoannot/man/XGR_filter_assays.html" class="external-link">XGR_filter_assays</a>).</p></dd>


<dt>adjust</dt>
<dd><p>The granularity of the peaks.</p></dd>


<dt>force_new</dt>
<dd><p>Download and prepare a new query
even if the file already exists locally (Default: <code>FALSE</code>).</p></dd>


<dt>show_plot</dt>
<dd><p>Print the plot.</p></dd>


<dt>nThread</dt>
<dd><p>Number of threads 
to parallelise downloading annotations over.</p></dd>


<dt>verbose</dt>
<dd><p>Print messages.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>List with the "data" and the "plot".</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">xgr_out</span> <span class="op">&lt;-</span> <span class="fu">echoplot</span><span class="fu">::</span><span class="fu">XGR_plot</span><span class="op">(</span>dat <span class="op">=</span> <span class="fu">echodata</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/echodata/man/BST1.html" class="external-link">BST1</a></span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">1000</span><span class="op">)</span>, <span class="op">]</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> echoannot:: Plotting XGR annotations.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Start at 2022-10-30 18:26:52</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 'ENCODE_TFBS_ClusteredV3_CellTypes' (from http://galahad.well.ox.ac.uk/bigdata/ENCODE_TFBS_ClusteredV3_CellTypes.RData) has been loaded into the working environment (at 2022-10-30 18:27:03)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> End at 2022-10-30 18:27:03</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Runtime in total is: 11 secs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting dat to GRanges object.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 398 query SNP(s) detected with reference overlap.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Ignoring unknown parameters: facets</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>no non-missing arguments to max; returning -Inf</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder, Jack Humphrey, Towfique Raj.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

